libs/chain-signatures/contract/tests/tee.rs

```diff
@@ -35,4 +36,80 @@ async fn test_tee_verify_no_tee() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_vote_code_hash() -> anyhow::Result<()> {
+    let (_, contract, accounts, _) = init_env_secp256k1(1).await;
+
+    let mpc_hash = MpcDockerImageHash::from([
+        0xab, 0xcd, 0xef, 0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef, 0x12, 0x34, 0x56, 0x78,
+        0x90, 0xab, 0xcd, 0xef, 0x12, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef, 0x12, 0x34, 0x56,
+        0x78, 0x90,
+    ]);
+    let args = serde_json::json!({"code_hash": mpc_hash});
+
+    // Helper function to get allowed hashes
+    let get_allowed_hashes = || async {
+        contract
+            .call("allowed_code_hashes")
+            .args_json(serde_json::json!(""))
+            .max_gas()
+            .transact()
+            .await?
+            .json::<Vec<MpcDockerImageHash>>()
+    };
+
+    // Helper function to get latest code hash
+    let get_latest_code_hash = || async {
+        contract
+            .call("latest_code_hash")
+            .args_json(serde_json::json!(""))
+            .max_gas()
+            .transact()
+            .await?
+            .json::<Option<MpcDockerImageHash>>()
+    };
+
+    // Helper function to vote on hash
+    let vote_hash = |account_idx: usize| {
+        let contract = &contract;
+        let accounts = &accounts;
+        let args = &args;
+        async move {
+            check_call_success(
+                accounts[account_idx]
+                    .call(contract.id(), "vote_code_hash")
+                    .args_json(args)
+                    .transact()
+                    .await?,
+            );
+            Ok::<(), anyhow::Error>(())
+        }
+    };
+
+    // First vote - should not be enough
+    vote_hash(0).await?;
+    assert_eq!(get_allowed_hashes().await?.len(), 0);
+    // Should get an error when no code hash is available yet
+    assert!(get_latest_code_hash().await.is_err());
+
+    // Second vote - should reach threshold
+    vote_hash(1).await?;
+    let allowed_hashes = get_allowed_hashes().await?;
+    assert_eq!(allowed_hashes, vec![mpc_hash.clone()]);
```

Comment:
before this check, it would be nice to maybe also test the behaviors:
- for when a participant changes their vote
- for when one participant votes for a different hash.